### ISSUE TITLE
Use Elasticsearch compatibility header with go-elasticsearch

### ISSUE THREAD

**Issue by Mpdreamz** [elastic/apm-server#6426]
Elasticsearch `7.10` introduces compatibility headers:

```
Accept:                 application/vnd.elasticsearch+json; compatible-with=7
Content-Type:           application/vnd.elasticsearch+json; compatible-with=7
```

Meaning Elasticsearch takes care of accepting and returning a `7.x` compatible response.

by utilizing this we can support running APM Server 7.16.x running against `Elasticsearch 8.x` with minimal accounting for breaking changes.

Compatible applies to format only, behavioral changes may occur. 

On 7.x the return type is still json:

```
content-type:       application/json; charset=UTF-8
```

On 8.x the return type matches the `Accept` header:

```
content-type:       application/vnd.elasticsearch+json;compatible-with=7
```

https://github.com/elastic/go-elasticsearch is setup to handle this but does not offer a nice configuration option yet (cc @Anaethelion)

**Comment by axw** [comment]
I see the following options:
 1. Enable the compatibility header by default, breaking support for older Elasticsearch versions. IMO this is not a reasonable option -- we explicitly support all 7.x.
 2. Provide users an option to enable the compatibility header, but default to off. Not great, but least complex.
 3. Have apm-server discover the Elasticsearch version, and automatically enable the header. Better for users if it works, but more complex.

I think (2) might even be possible today, by setting `output.elasticsearch.headers`.

**Comment by simitt** [comment]
I tested option (2) to let users decide to configure compatibility with ES 8.0.

With current libbeat behavior I ran into issues where the `Acceot` and `Content-Type` headers were always also set by libbeat, leading to invalid headers. A PR to change the current behavior and support the custom headers is up https://github.com/elastic/beats/pull/29176. 

With https://github.com/elastic/go-elasticsearch/pull/355 the go-elasticsearch client introduced a config setting `enable_compatibility_mode` that can be used to enable the compatibility header when working with the `v7` client. This requires some changes on the APM Server side, to ensure the config option is passed on. The go-es-client does not allow the `accept` and `content-type` headers to be explicitly configured, as it would add the headers leading to invalid headers. PR is up https://github.com/elastic/apm-server/pull/6742
